### PR TITLE
Fix #196, Fix timer-test to be able to use OS_MAX_TIMERS value

### LINKED ISSUE #196: Functional Timer Test Hard Codes Configuration Value
The functional timer test (/src/tests/timer-test/timer-test.c) hard codes the number of timers to 4, which may be higher than the OS_MAX_TIMERS configuration set in osconfig.h.

It is recommended to update the test to use the OS_MAX_TIMERS configuration macro or add protection with an #if guard.

### PR DESCRIPTION
**Pull request by yammajamma**
Fix #196, Fix timer-test to be able to use OS_MAX_TIMERS value

**Describe the contribution**
Changed timer-test to be able to use OS_MAX_TIMERS value on top of the hard-coded NUMBER_OF_TIMERS value.  This will allow the test to be functional even if the OS_MAX_TIMERS value is reconfigured.

**Testing performed**
Steps taken to test the contribution:
1. Standard build and ran unit tests.

**System(s) tested on**
cFS Dev Server
OS: Ubuntu 18.04
Versions: OSAL 5.0.11.0

**Contributor Info - All information REQUIRED for consideration of pull request**
Yasir Majeed Khan
NASA GSFC/Emergent Space

### CODE REVIEW

**Comment by yammajamma** [comment]
@skliper, @jphickey  would you mind reviewing this when you get a chance? Thanks!

File: src/tests/timer-test/timer-test.c [hunk 1]
```diff
@@ -20,24 +20,29 @@
...
+
 void TimerTestSetup(void);
 void TimerTestTask(void);
 void TimerTestCheck(void);
 
 OS_time_t        StartTime;
 OS_time_t        EndTime;
+uint32           TimerStart[NUMBER_OF_TIMERS];
+uint32           TimerInterval[NUMBER_OF_TIMERS]; 
+
```
Comment (jphickey) [inline src/tests/timer-test/timer-test.c:32]: This looks odd - variable declared without initializer and then defined with initializer.  Possibly a remnant change?  Should remove this addition.

Comment (yammajamma) [reply src/tests/timer-test/timer-test.c:32]: This was a typo.  Fixed.

**Comment by yammajamma** [comment]
@jphickey Yes, I tested it with less than 4 timers, etc. and it worked as expected.  I'm pushing the other change now.

**Comment by astrogeco** [comment]
CCB-20200624 APPROVED

**Pull request merged**